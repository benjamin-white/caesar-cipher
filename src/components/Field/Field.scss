.field {
  align-items: center;
  display: flex;
  margin-bottom: 1rem;

  &-label,
  &-input {
    width: 200px;
  }

  &-input {
    border: 1px solid rgba(0 0 0 / 0.25);
    border-radius: 0.125rem;
    padding: 0.25rem 0.5rem;

    &:focus {
      outline: 1px solid rgba(0 0 0 / 0.25);
    }
  }
}
