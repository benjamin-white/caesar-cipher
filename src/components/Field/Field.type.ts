export type FieldProps = {
  children: (className: string, id: string) => React.ReactNode;
  labelText: string;
  id: string;
};
