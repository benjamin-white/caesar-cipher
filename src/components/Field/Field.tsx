import { type FieldProps } from './Field.type';
import './Field.scss';

const Field = ({ children, labelText, id }: FieldProps) => (
  <div className="field">
    <label className="field-label" htmlFor={id}>
      {labelText}
    </label>
    {children('field-input', id)}
  </div>
);

export default Field;
