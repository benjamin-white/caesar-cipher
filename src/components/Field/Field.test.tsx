import { render, screen } from '@testing-library/react';
import Field from './';

describe('Field', () => {
  it('renders a `label` and child component', () => {
    const { container } = render(
      <Field labelText="Button label" id="button">
        {(className, id) => (
          <button type="button" className={className} id={id}>
            Button
          </button>
        )}
      </Field>
    );

    const label = screen.queryByText('Button label');
    const child = screen.getByLabelText('Button label');

    expect(label?.getAttribute('for')).toEqual(child.id);
    expect(child).toHaveClass('field-input');
    expect(container).toMatchSnapshot();
  });
});
