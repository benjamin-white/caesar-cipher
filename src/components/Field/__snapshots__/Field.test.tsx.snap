// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Field renders a \`label\` and child component 1`] = `
<div>
  <div
    class="field"
  >
    <label
      class="field-label"
      for="button"
    >
      Button label
    </label>
    <button
      class="field-input"
      id="button"
      type="button"
    >
      Button
    </button>
  </div>
</div>
`;
